components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,303 @@
+@PowerMockIgnore("java.sql.*")
+public class SCIMUserManagerTest extends PowerMockTestCase {
+
+    @Mock
+    private UserStoreManager mockedUserStoreManager;
+
+    @Mock
+    private ClaimManager mockedClaimManager;
+
+    @Mock
+    private GroupDAO mockedGroupDAO;
+
+    @Mock
+    private SCIMAttributeSchema mockedScimAttributeSchema;
+
+    @Mock
+    private RealmConfiguration mockedRealmConfig;
+
+    @Mock
+    private User mockedUser;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+       initMocks(this);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testCreateUser() throws Exception {
+
+    }
+
+    @Test
+    public void testGetUser() throws Exception {
+    }
+
+    @Test
+    public void testDeleteUser() throws Exception {
+    }
+
+    @Test
+    public void testListUsersWithGET() throws Exception {
+    }
+
+    @Test
+    public void testListUsersWithPost() throws Exception {
+    }
+
+    @Test
+    public void testUpdateUser() throws Exception {
+    }
+
+    @DataProvider(name = "data")
+    public Object[][] data() {
+
+        String CoreClaimUri1;
+        String testMappedAttributesCore1;
+
+        String CoreClaimUri2;
+        String testMappedAttributesCore2;
+
+        String UserClaimUri1;
+        String testMappedAttributesUser1;
+
+        String UserClaimUri2;
+        String testMappedAttributesUser2;
+
+        Claim claim1 = new Claim();
+        Claim claim2 = new Claim();
+        Claim claim3 = new Claim();
+        Claim claim4 = new Claim();
+
+        CoreClaimUri1 =  "testCoreClaimURI1";
+        claim1.setClaimUri(CoreClaimUri1);
+        testMappedAttributesCore1 = "MappedAttributesCore1";
+
+        CoreClaimUri2 =  "testCoreClaimURI2";
+        claim2.setClaimUri(CoreClaimUri2);
+        testMappedAttributesCore2 = "MappedAttributesCore2";
+
+        UserClaimUri1 =  "testUserClaimURI1";
+        claim3.setClaimUri(UserClaimUri1);
+        testMappedAttributesUser1 = "MappedAttributesUser1";
+
+        UserClaimUri2 =  "testUserClaimURI2";
+        claim4.setClaimUri(UserClaimUri2);
+        testMappedAttributesUser2 = "MappedAttributesUser2";
+
+        ClaimMapping cMap1 = new ClaimMapping(claim1, testMappedAttributesCore1);
+        ClaimMapping cMap2 = new ClaimMapping(claim2, testMappedAttributesCore2);
+        ClaimMapping cMap3 = new ClaimMapping(claim3, testMappedAttributesUser1);
+        ClaimMapping cMap4 = new ClaimMapping(claim4, testMappedAttributesUser2);
+
+        ClaimMapping[] coreClaims = new ClaimMapping[]{cMap1, cMap2, cMap3, cMap4};
+
+        HashMap<String,Boolean> requiredAttributes = new HashMap<String, Boolean>() {
+            {
+                put("test1.test",true);
+            }
+        };
+
+        String[] roles = new String[]{"role1","role2","role3"};
+
+        return new Object[][]{
+                {coreClaims,requiredAttributes,roles}
```

Comment:
can use the data directly if there is only one parameter set.

Reply:
Also there are formatting issues

Reply:
More data sets need to be added to cover all the branches. So shall I keep the data provider here?